### PR TITLE
Tidy up variant analysis commands

### PR DESCRIPTION
**Pull request by charisk**
Paired with @robertbrignull.

The `codeQL.runVariantAnalysis` and `runVariantAnalysisContextEditor` commands were both calling the same underlying code, making types a bit hard to follow. This makes the commands and types more explicit early on
 
## Checklist
N/A:
- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by robertbrignull — APPROVED** [review]
LGTM but obviously we paired on it, so I'll leave it up to you if you want to get another opinion.

**Review by shati-patel — APPROVED** [review]
One comment/question, otherwise LGTM

File: extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts [hunk 1]
```diff
@@ -184,44 +184,53 @@ export class VariantAnalysisManager
...
-        ),
-      {
-        title: "Run Variant Analysis",
-        cancellable: true,
-      },
-    );
+  private async runVariantAnalysisFromCommandPalette() {
+    const fileUri = Window.activeTextEditor?.document.uri;
+    if (!fileUri) {
+      throw new Error("Please select a .ql file to run variant analysis on");
```
Comment (shati-patel) [inline extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts:190]: Slightly nit-picky comment: This sounds like we're running a variant analysis on a `.ql` file, which doesn't really make sense 🙃 

Maybe something like:

```suggestion
      throw new Error("Please select a .ql file to run as a variant analysis");
```
(feel free to ignore though!)
